lib/smart_answer/date_range.rb

```diff
@@ -0,0 +1,32 @@
+module SmartAnswer
+  class DateRange
+    LATEST_DATE = Date::Infinity.new
+
+    attr_reader :begins_on, :ends_on
+
+    def initialize(begins_on:, ends_on: LATEST_DATE, duration: nil)
+      @begins_on = begins_on
+      @ends_on = duration.present? ? (begins_on + duration - 1) : ends_on
+    end
+
+    def include?(date)
+      (date >= @begins_on) && (date <= @ends_on)
+    end
+
+    def number_of_days
+      (@ends_on == LATEST_DATE) ? Float::INFINITY : (@ends_on - @begins_on).to_i + 1
+    end
+
+    def ==(other)
+      eql?(other)
+    end
+
+    def eql?(other)
+      (self.class == other.class) && ([begins_on, ends_on] == [other.begins_on, other.ends_on])
+    end
+
+    def hash
```

Comment:
That method probably isn't being used at the moment, but since I've overridden `#eql?`, I think I'm duty-bound to override `#hash` to make it consistent. I want `DateRange` to be a proper immutable [ValueObject](http://c2.com/cgi/wiki?ValueObject) and this feels like the right thing to do to avoid surprises later. The implementation of these equality-related methods was taken from the [Values gem](https://github.com/tcrayford/Values).